clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -863,17 +873,35 @@ class KernelObjVisitor {
     QualType ET = CAT->getElementType();
     int64_t ElemCount = CAT->getSize().getSExtValue();
     std::initializer_list<int>{(handlers.enterArray(), 0)...};
-    for (int64_t Count = 0; Count < ElemCount; Count++) {
-      VisitElement(nullptr, FD, ET, handlers...);
+
+    if (ElemCount > 0) {
```

Comment:
Isn't it always true?

Reply:
Nope, not necessarily apparently: https://godbolt.org/z/eox5ME

Its rare, but possible.

Reply:
Is it zero-sized array? Didn't we prohibit them for SYCL?

Reply:
Looks like you're right!  Thanks!